File: android/app/src/main/java/io/nearpay/sdk/NearPay.java
package io.nearpay.sdk;

import android.content.Context;
import io.nearpay.sdk.data.models.AuthenticationData;
import io.nearpay.sdk.utils.enums.NetworkConfiguration;
import io.nearpay.sdk.utils.enums.UIPosition;
import io.nearpay.sdk.utils.listeners.PaymentListener;
import java.util.Locale;

// STUB CLASS - Replace with real NearPay SDK
public class NearPay {
    
    public void purchase(Double amount, PaymentListener listener) {
        // Stub implementation - will be replaced by real SDK
        android.util.Log.w("NearPay", "STUB SDK: Real NearPay SDK not integrated yet");
    }

    public static class Builder {
        public Builder context(Context context) { return this; }
        public Builder authenticationData(AuthenticationData data) { return this; }
        public Builder environment(Environments env) { return this; }
        public Builder locale(Locale locale) { return this; }
        public Builder networkConfiguration(NetworkConfiguration config) { return this; }
        public Builder uiPosition(UIPosition position) { return this; }
        public Builder loadingUi(boolean show) { return this; }
        public NearPay build() { return new NearPay(); }
    }
}

File: android/app/src/main/java/io/nearpay/sdk/NearPay.kt
package io.nearpay.sdk

import android.app.Activity
import io.nearpay.sdk.data.models.AuthenticationData
import io.nearpay.sdk.utils.enums.NetworkConfiguration
import io.nearpay.sdk.utils.enums.UIPosition
import io.nearpay.sdk.utils.listeners.SetupListener
import io.nearpay.sdk.utils.listeners.PurchaseListener
import java.util.Locale

class NearPay private constructor() {
    
    fun setup(listener: SetupListener) {
        // Stub: Real implementation requires NearPay Direct SDK
    }
    
    fun purchase(amount: Long, listener: PurchaseListener) {
        // Stub: Real implementation requires NearPay Direct SDK
    }
    
    class Builder {
        private var activity: Activity? = null
        private var authenticationData: AuthenticationData? = null
        private var environment: Environments? = null
        private var locale: Locale? = null
        private var networkConfiguration: NetworkConfiguration? = null
        private var uiPosition: UIPosition? = null
        
        fun context(activity: Activity) = apply { this.activity = activity }
        fun authenticationData(data: AuthenticationData) = apply { this.authenticationData = data }
        fun environment(env: Environments) = apply { this.environment = env }
        fun locale(locale: Locale) = apply { this.locale = locale }
        fun networkConfiguration(config: NetworkConfiguration) = apply { this.networkConfiguration = config }
        fun uiPosition(position: UIPosition) = apply { this.uiPosition = position }
        
        fun build(): NearPay {
            return NearPay()
        }
    }
}
